### PR TITLE
Adding team based BLAS functions

### PR DESCRIPTION
**Pull request by vqd8a**
Addressing issue #209

### CODE REVIEW

**Comment by mhoemmen** [comment]
The `abs` function does not appear anywhere in this code -- may we rename this PR?

File: src/blas/KokkosBlas1_team_axpby.hpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#define KOKKOSBLAS1_TEAM_AXPBY_HPP_
+
+#include<KokkosBlas1_team_axpby_spec.hpp>
+
+namespace KokkosBlas {
+namespace Experimental {
+
+template<class TeamType, class XVector,class YVector>
+void KOKKOS_INLINE_FUNCTION axpby (const TeamType& team,
+                 const typename XVector::non_const_value_type& a, const XVector& x,
```
[the remark this answers is not in the document]
Comment (crtrott) [reply src/blas/KokkosBlas1_team_axpby.hpp]: More templating ...

**Comment by vqd8a** [comment]
Still working on unit tests.

File: src/blas/KokkosBlas1_team_axpby.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (crtrott) [reply src/blas/KokkosBlas1_team_axpby.hpp]: Also with the Scalars being dependent types, you should actually get implicit conversion rules for this. So your example should just work.

File: src/blas/KokkosBlas2_team_gemv.hpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                 const XVector& x,
+                 const typename YVector::non_const_value_type& beta,
+                 const YVector& y)
+{
+  if (trans[0] == 'N' || trans[0] == 'n')
+    return Impl::TeamGEMV<TeamType,MatrixType,XVector,YVector,0>::team_gemv(team,alpha,A,x,beta,y);
+  if (trans[0] == 'T' || trans[0] == 't')
+    return Impl::TeamGEMV<TeamType,MatrixType,XVector,YVector,1>::team_gemv(team,alpha,A,x,beta,y);
+  if (trans[0] == 'C' || trans[0] == 'c')
+    return Impl::TeamGEMV<TeamType,MatrixType,XVector,YVector,1>::team_gemv(team,alpha,A,x,beta,y);
```
[the remark this answers is not in the document]
Comment (crtrott) [reply src/blas/KokkosBlas2_team_gemv.hpp]: he actually based this on my branch in the other pull request which adds the team_gemv, team_axpby and team_dot. I wrote those ...

[the remark this answers is not in the document]
Comment (vqd8a) [reply src/blas/KokkosBlas2_team_gemv.hpp]: Yes, I am about to say this.

[the remark this answers is not in the document]
Comment (crtrott) [reply src/blas/KokkosBlas2_team_gemv.hpp]: And yes the C case needs to have a 2 as the template parameter ...

**Comment by vqd8a** [comment]
Added team-based abs (+ unit tests for team-based abs, dot, axpby, axpy, and gemv) for Serial, OpenMP, Cuda, Pthread.

**Comment by vqd8a** [comment]
Addressing issue #214

**Comment by crtrott** [comment]
This looks already pretty good. I will do a proper code review later but this is outstanding progress!

**Comment by crtrott** [comment]
Uhm what happened?

**Comment by vqd8a** [comment]
?I just wanted to rename the branch issue-209.


How can I recover the pull request #210?

________________________________

**Comment by crtrott** [comment]
Not sure you can. Just issue a new pull request. Generally renaming branches is something to be avoided. But its not a big deal.

**Comment by vqd8a** [comment]
Oops. Sorry

**Comment by crtrott** [comment]
Yeah its one of those things one learns the hard way 😏
In this case it really doesnt matter but when multiple people work on the same branch such a renaming can cause some havoc.

**Comment by vqd8a** [comment]
@crtrott Thanks. Just created PR #215.

**Pull request closed without merging**